include/arch/arc/arch.h

```diff
@@ -79,87 +80,70 @@ extern "C" {
 		1 << (31 - __builtin_clz(x) + 1) : \
 		1 << (31 - __builtin_clz(x)))
 
-#if defined(CONFIG_USERSPACE)
 
+#if defined(CONFIG_USERSPACE)
 #define Z_ARCH_THREAD_STACK_RESERVED \
 	(STACK_GUARD_SIZE + CONFIG_PRIVILEGED_STACK_SIZE)
+#else
+#define Z_ARCH_THREAD_STACK_RESERVED (STACK_GUARD_SIZE)
+#endif
 
-#if CONFIG_ARC_MPU_VER == 2
 
-#define Z_ARCH_THREAD_STACK_DEFINE(sym, size) \
-	struct _k_thread_stack_element __noinit \
-		__aligned(POW2_CEIL(STACK_SIZE_ALIGN(size))) \
-		sym[POW2_CEIL(STACK_SIZE_ALIGN(size)) + \
-		+  STACK_GUARD_SIZE + CONFIG_PRIVILEGED_STACK_SIZE]
+/* MPUv2 requires size must be power of 2 and >= 2048 */
+#define Z_ARCH_THREAD_STACK_SIZE(size) POW2_CEIL(STACK_SIZE_ALIGN(size))
 
+#if  defined(CONFIG_USERSPACE) && CONFIG_ARC_MPU_VER == 2
```

Comment:
Please add a comment on rationale/explanation  for the alignment requirements: user stack and guard are protected using MPU regions, so need to adhere to MPU start, size alignment.
And something similar for the no-userspace guard below: kernel stacks are not MPU protected, only guard needs to be aligned.

Reply:
done